lib/diagnostic.rb

```diff
@@ -1,10 +1,12 @@
 # Ruby enumerable diagnostic
 class Diagnostic
+
   # Question 1
   # Write ruby code to return the longest entry in the
   # enumerable supplied.  Assume each entry implements a length
   # method that returns a Fixnum.
   def longest(enum)
+    enum.reduce(0) { |m, o| m = m.length < o.length? o : m }
```

Comment:
This will _almost_ work - passing in 0 as a starting value will actually break the `reduce`, since `0` doesn't have a `.length` method. If you didn't include a starting value, it would just begin with the entry 0 and entry 1.